### DISCUSSION TITLE
How to access points from o3d.t.io.read_point_cloud?

### DISCUSSION THREAD

**Discussion by nimafo** [isl-org/Open3D#6366] [Q&A]
Dear community,
I am trying to access points in the form of Numpy XYZ array from the following block of code for the main points cloud(`point_cloud`) and the boundary points(`boundaries`) which have the same data structure. How can I do that?
```
import open3d as o3d

# Load or create a point cloud
ply_file_path = 'cluster_4.ply'
point_cloud = o3d.t.io.read_point_cloud(ply_file_path)
point_cloud.estimate_normals()
radius = 0.05
max_nn = 15
boundaries, mask = point_cloud.compute_boundary_points(radius, max_nn)
```

when running the variables I get the following:

> PointCloud on CPU:0 [74919 points (Float32)].
> Attributes: normals (dtype = Float32, shape = {74919, 3}).

**Comment by saurabheights** [comment]
If you are trying to convert from numpy to PointCloud, you can try:-

http://www.open3d.org/docs/release/python_api/open3d.utility.Vector3dVector.html

Note: you can add color of pointcloud in same way as `points`.

```
import open3d
import numpy as np

pcd = open3d.geometry.PointCloud()
np_points = np.random.rand(100, 3)
np_colors = np.random.rand(100, 3)  

# From numpy to Open3D
pcd.points = open3d.utility.Vector3dVector(np_points)
pcd.colors = open3d.utility.Vector3dVector(np_colors)

open3d.visualization.draw(pcd)
```

Result:-

![image](https://github.com/isl-org/Open3D/assets/11634769/2a4c0e63-035c-420c-9596-85f64501e53e)

**Reply by nimafo** [reply]
I am actually trying to convert `o3d.t.io.read_point_cloud(ply_file_path)` to numpy array.

**Reply by ssheorey** [reply]
HI @nimafo you can use the `numpy()` function on Tensors to convert from an Open3D tensor to a Numpy array.

```py
# Access the `positions` Tensor property of the `point` data structure in a point cloud.
npy_pts = point_cloud.point.positions.numpy()   
```

http://www.open3d.org/docs/latest/python_api/open3d.core.Tensor.html?highlight=numpy#open3d.core.Tensor.numpy
